modules/xml/src/smithy4s/xml/XmlDocument.scala

```diff
@@ -47,11 +52,11 @@ object XmlDocument {
     * ADT are produced
     */
   // format: off
-  sealed trait XmlContent
-  case class XmlText(text: String)                                                       extends XmlContent
-  case class XmlElem(name: XmlQName, attributes: List[XmlAttr], children: List[XmlContent]) extends XmlContent
-  case class XmlAttr(name: XmlQName, value: List[XmlText])
-  case class XmlQName(prefix: Option[String], name: String) {
+  sealed trait XmlContent extends Product with Serializable
+  final case class XmlText(text: String)                                                       extends XmlContent
+  final case class XmlElem(name: XmlQName, attributes: List[XmlAttr], children: List[XmlContent]) extends XmlContent
+  final case class XmlAttr(name: XmlQName, value: List[XmlText]) extends XmlContent
```

Comment:
this really helps simplify the encoding logic, at the cost of being a little bit less correct, modelling wise

Reply:
minor: `values` ?

Reply:
[d45899d](https://github.com/disneystreaming/smithy4s/pull/503/commits/d45899dc33844c1b601d57e2076470c2af4b1a94)